block/manager.go

```diff
@@ -237,21 +237,27 @@ func (m *Manager) onNodeHealthStatus(event pubsub.Message) {
 // TODO: move to gossip.go
 // onNewGossippedBlock will take a block and apply it
 func (m *Manager) onNewGossipedBlock(event pubsub.Message) {
-	m.retrieverMutex.Lock() // needed to protect blockCache access
 	eventData := event.Data().(p2p.GossipedBlock)
 	block := eventData.Block
 	commit := eventData.Commit
-	m.logger.Debug("Received new block via gossip", "height", block.Header.Height, "n cachedBlocks", len(m.blockCache))
+	m.retrieverMutex.Lock() // needed to protect blockCache access
+	_, found := m.blockCache[block.Header.Height]
+	if found {
+		m.retrieverMutex.Unlock()
+		return
+	}
```

Comment:
can you add a comment here explaining why we return early?
Like your explanation here https://github.com/dymensionxyz/dymint/pull/818#issuecomment-2103415147
Because otherwise, to a new reader, it probably looks like a useless branch

Reply:
we return early if the block is already received, since it is not necessary to cache again the same block (and therefore avoid attempting applying blocks if actually no new block is received).